packages/backend/src/services/UnfurlService/UnfurlService.ts

```diff
@@ -590,6 +597,52 @@ export class UnfurlService extends BaseService {
                     });
                     return imageBuffer;
                 } catch (e) {
+                    /**
+                     * An error can be retried if it is a TimeoutError or a ProtocolError
+                     * - TimeoutError: when the page is not loaded in time or the element is not found
+                     * - ProtocolError: when the page is closed before the screenshot is taken
+                     * - That include the message "Target closed" or "Protocol error" - e.g: Protocol error (Page.captureScreenshot): Target closed or Protocol error (DOM.describeNode): Target closed"
+                     */
+                    const isRetryableError =
+                        e instanceof TimeoutError ||
+                        e instanceof ProtocolError ||
+                        e.message.includes('Protocol error') ||
+                        e.message.includes('Target closed');
+
+                    if (isRetryableError && retries) {
+                        this.logger.info(
+                            `Retrying: unable to fetch screenshots for scheduler with url ${url}, of type: ${lightdashPage}. Message: ${e.message}`,
+                        );
+                        span.recordException(e);
+                        span.setAttributes({
+                            'page.type': lightdashPage,
+                            url,
+                            chartType: chartType || 'undefined',
+                            organization_uuid: organizationUuid || 'undefined',
+                            uuid: resourceUuid ?? 'undefined',
+                            title: resourceName ?? 'undefined',
+                            is_retrying: true,
+                            custom_width: `${gridWidth}`,
+                        });
+                        span.setStatus({
+                            code: SpanStatusCode.ERROR,
+                        });
```

Comment:
Will it still be marked as error if it succeeds on second try?

Reply:
Is going to create `UnfurlService.saveScreenshot` nested spans and the second one should show as success 👍